### PR TITLE
CLOUDP-262694: Fix dockerNotFound error message.

### PR DESCRIPTION
**Pull request by jeroenvervaeke**
## Proposed changes
- Fixed docker error message
- Fixed post_run_opts to:
    - use container engine, get rid of reference to podman
    - show message if docker or podman is not installed, not only podman
- Fix broken tests

_Jira ticket:_ CLOUDP-262694

### CODE REVIEW

File: internal/container/docker.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ import (
 	"time"
 )
 
-var ErrDockerNotFound = errors.New("podman not found in your system, check requirements at https://dochub.mongodb.org/core/atlas-cli-deploy-local-reqs")
+var ErrDockerNotFound = fmt.Errorf("%w: docker not found in your system, check requirements at https://dochub.mongodb.org/core/atlas-cli-deploy-local-reqs", ErrContainerEngineNotFound)
```
Comment (fmenezes) [inline internal/container/docker.go:29]: [q] instead of joining the errors here how about doing it on `container_new_linux.go` and `container_new.go`?

Comment (jeroenvervaeke) [reply internal/container/docker.go:29]: What would be the advantage of doing that?
Keep in mind that on some linux repos we'll have both linux/podman

Comment (fmenezes) [reply internal/container/docker.go:29]: That is exactly why, one place to wrap the error, I haven't seen the pattern of fmt.Errorf in a package level yet.

It is normally done on functions.

**Review by fmenezes — APPROVED** [review]
I'm ok with the change, my comments are nits

**Pull request merged**